src/test/demultiplexer_node_test.cpp

```diff
@@ -0,0 +1,87 @@
+        DLNode(nodeName, modelName, modelVersion, modelManager, nodeOutputNameAlias, demultiplyCount.value_or(0)) {
+        this->demultiplexCount = demultiplexCount;
+        // createSession to have source session for fetchResults()
+        std::unique_ptr<NodeSession> nodeSession = createNodeSession(meta, 1);
+        auto emplacePair = nodeSessions.emplace(meta.getSessionKey(), std::move(nodeSession));
+        EXPECT_TRUE(emplacePair.second);
+    }
+    void setFetchResult(InferenceEngine::Blob::Ptr& blobToReturn) {
+        this->blobToReturn = blobToReturn;
+    }
+    using Node::fetchResults;
+    Status fetchResults(NodeSession& nodeSession, SessionResults& nodeSessionOutputs) {
+        const auto& sessionMetadata = nodeSession.getNodeSessionMetadata();
+        const auto sessionKey = sessionMetadata.getSessionKey();
+        BlobMap blobs{{std::string("a"), blobToReturn}};
+        std::pair<NodeSessionMetadata, BlobMap> metaBlobsPair{sessionMetadata, blobs};
+        nodeSessionOutputs.emplace(sessionKey, std::move(metaBlobsPair));
+        return demultiplyOutputs(nodeSessionOutputs);
+    }
+
+private:
+    InferenceEngine::Blob::Ptr blobToReturn;
+};
+
+TEST(DemultiplexerTest, CheckDemultipliedBlobs) {
+    const uint16_t demultiplyCount = 2;
+    // prepare pre demultiplexer blob
+    std::vector<float> blobData1{-1, 4, 5, 12, 3, 52, 12, 0.5, 9, 1.67, 0, 8};
+    std::vector<float> blobData2{4, 42, 35, -2, 13, 2, -1, 0.9, -0.3, 4.67, 100, 80};
+    const std::vector<size_t> shape{1, demultiplyCount, blobData1.size()};
+    const InferenceEngine::Precision precision{InferenceEngine::Precision::FP32};
+    const InferenceEngine::Layout layout{InferenceEngine::Layout::CHW};
+    const InferenceEngine::TensorDesc desc{precision, shape, layout};
+    std::vector<float> blobDataNonDemultiplexed(blobData1.size() * demultiplyCount);
+    std::copy(blobData1.begin(), blobData1.end(), blobDataNonDemultiplexed.begin());
+    std::copy(blobData2.begin(), blobData2.end(), blobDataNonDemultiplexed.begin() + blobData1.size());
+    InferenceEngine::Blob::Ptr inputBlob = InferenceEngine::make_shared_blob<float>(desc, blobDataNonDemultiplexed.data());
+    // construct demultiplexer node
+    NodeSessionMetadata meta;
+    ConstructorEnabledModelManager manager;
+    DemultiplexerDLNode demultiplexerNode("node", "model", 1, manager, std::unordered_map<std::string, std::string>{{"a", "a"}}, demultiplyCount, meta);
+    demultiplexerNode.setFetchResult(inputBlob);
+    SessionResults sessionResults;
+    session_key_t sessionKey = meta.getSessionKey();
+    // perform test
+    auto status = demultiplexerNode.fetchResults(sessionKey, sessionResults);
+    ASSERT_EQ(status, StatusCode::OK);
+    ASSERT_EQ(sessionResults.size(), 2);
+    auto& sessionResult1 = sessionResults["node_0"];
```

Comment:
Replace using "node_0" & "node_1" with demultiplexedMetadata[0/1].

auto demultiplexedMetadata = meta.generateSubsession("node", demultiplyCount);

You could as well create demultiplexerNodeName = "node" and then reuse it in L71.